### PR TITLE
Implement store for heap allocated tensors.

### PR DESCRIPTION
**Pull request by geirst**
@toregge please review
@arnej27959 FYI

### CODE REVIEW

File: searchlib/src/tests/tensor/direct_tensor_store/direct_tensor_store_test.cpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include <vespa/searchlib/tensor/direct_tensor_store.h>
+#include <vespa/vespalib/gtest/gtest.h>
+#include <vespa/eval/tensor/default_tensor_engine.h>
+#include <vespa/eval/tensor/tensor.h>
+
+#include <vespa/log/log.h>
+LOG_SETUP("direct_tensor_store_test");
```
Comment (toregge) [inline searchlib/src/tests/tensor/direct_tensor_store/direct_tensor_store_test.cpp:9]: Consider removing unneeded log setup.

File: searchlib/src/tests/tensor/direct_tensor_store/direct_tensor_store_test.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    virtual ~DirectTensorStoreTest() {
+        store.clearHoldLists();
+    }
+
+    void expect_tensor(const Tensor* exp, EntryRef ref) {
+        const auto* act = store.get_tensor(ref);
+        ASSERT_TRUE(act);
+        EXPECT_EQ(exp, act);
+        EXPECT_EQ(*exp, *act);
```
Comment (toregge) [inline searchlib/src/tests/tensor/direct_tensor_store/direct_tensor_store_test.cpp:50]: Consider removing unneeded equality check, passing identity check implies equality.

**Review by toregge — APPROVED** [review]

File: searchlib/src/tests/tensor/direct_tensor_store/direct_tensor_store_test.cpp (hunk 1 above)
Comment (geirst) [reply searchlib/src/tests/tensor/direct_tensor_store/direct_tensor_store_test.cpp]: Fixed

File: searchlib/src/tests/tensor/direct_tensor_store/direct_tensor_store_test.cpp (hunk 2 above)
Comment (geirst) [reply searchlib/src/tests/tensor/direct_tensor_store/direct_tensor_store_test.cpp]: Fixed

**Review by toregge — APPROVED** [review]

**Pull request merged**